test/Microsoft.ML.Tests/Transformers/ValueMappingTests.cs

```diff
@@ -89,6 +89,9 @@ public void ValueMapInputIsVectorTest()
             var estimator = new WordTokenizingEstimator(Env, new[]{
                     new WordTokenizingTransformer.ColumnInfo("TokenizeA", "A")
                 }).Append(new ValueMappingEstimator<ReadOnlyMemory<char>, int>(Env, keys, values, new[] { ("VecD", "TokenizeA"), ("E", "B"), ("F", "C") }));
+            var schema = estimator.GetOutputSchema(SchemaShape.Create(dataView.Schema));
+            Assert.True(schema.TryFindColumn("VecD", out var originalColumn));
```

Comment:
Could you add a `TestEstimatorCore`, to your tests for `ValueMap`? Should be pretty quick and I think it would help make sure there are no further inconsistencies.

Reply:
hmm... none of the test in this file use `TestEstimatorCore`. So, I would avoid that here too.
Also, checks made here are explicitly testing specific logic that I think cannot be tested with `TestEstimatorCore`.

---
In reply to: [253658781](https://github.com/dotnet/machinelearning/pull/2386#discussion_r253658781) [](ancestors = 253658781)

Reply:
The reason I think it's important to have `TestEstimatorCore` is that it checks that the schema in` GetOutpuSchema` matches the schema that is actually produced. Since you found a bug in the output schema, I think it would be good to have this special case of an vector input go through the `TestEstimatorCore`. There is a test that uses `TestEstimatorCore` bellow: `ValueMappingWorkout`. 

---
In reply to: [253661005](https://github.com/dotnet/machinelearning/pull/2386#discussion_r253661005) [](ancestors = 253661005,253658781)

Reply:
@zeahmed please see here:
https://github.com/dotnet/machinelearning/blob/2dfc51aa7172199c1b2145e48ecb9eb855bb87f1/test/Microsoft.ML.Tests/Transformers/ValueMappingTests.cs#L506

You can add to this test by adding a new DataView that produces an array. Then the TestEstimatorCore would validate as what @artidoro  is suggesting.

Reply:
Thanks for the explanation @artidoro and @singlis, I think this is an end to end scenario where an input is a valid input and output is being tested against expected output. The semantics of `TestEstimatorCore` is different where it test the schema propagation for valid, invalid inputs for estimators and transforms. 

Adding that here in this test needlessly complicate the test and will not test the problem that is being tested between line 92-94.

I think it should be done as a separate test and that is what `ValueMappingWorkout` is doing. However, its not doing it for all the input scenarios. So, I am creating an issue to do it separately https://github.com/dotnet/machinelearning/issues/2410